Application/Queries/PersonalCompensationsQuery.cs

```diff
@@ -0,0 +1,28 @@
+using Application.Queries.Contracts;
+using Microsoft.EntityFrameworkCore;
+using DataAccess;
+using Core;
+
+namespace Application.Queries;
+
+public class PersonalCompensationsQuery : IPersonalCompensationsQuery
+{
+    private readonly CompensationsDbContext _context;
+
+    private readonly ITenantCompensationsQuery _tenantCompensationsQuery;
+
+    public PersonalCompensationsQuery(CompensationsDbContext context, ITenantCompensationsQuery tenantCompensationsQuery)
+    {
+        _context = context;
+        _tenantCompensationsQuery = tenantCompensationsQuery;
+    }
+
+    public async Task<List<Compensation>> GetPersonalCompensationsAsync(long employeeId, long tenantId)
+    {
+        var compensations = await _tenantCompensationsQuery.GetCompensationsByTenantIdAsync(tenantId);
+
+        return compensations
+            .Where(x => x.EmployeeId == employeeId)
```

Comment:
You will need to revert code back to where we left together. Here is also on interesting nuance. You made a call to DB and it returned to you data for all employees and you filter by employeeId in memory of the process but not in db. It should be filtered on the db level (be a part of the expression before you are calling ToListAsync).